src/main/java/org/jenkinsci/plugins/nomad/NomadCloud.java

```diff
@@ -341,7 +361,14 @@ public Node call() throws Exception {
             String jnlpSecret = JnlpAgentReceiver.SLAVE_SECRET.mac(workerName);
 
             LOGGER.log(Level.INFO, "Asking Nomad to schedule new Jenkins worker");
-            nomad.startWorker(workerName, jnlpSecret, template);
+
+            String workerJob = nomad.startWorker(workerName, jnlpSecret, template);
+            JSONObject workerJobJSON = new JSONObject(workerJob).getJSONObject("Job");
+            String namespace = workerJobJSON.optString("Namespace");
+            if (!namespace.equals("")) {
+                   worker.setNamespace(namespace);
+            }
+            worker.setRegion(workerJobJSON.optString("Region"));
```

Comment:
Maybe we should have a way to represent jobs as an objects, and serialize them on demand before sending them to the API, instead of building strings and parsing them out again from JSON to get/change some values.